doc/alertmanager-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,135 @@
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'An Alertmanager instance failed to send notifications.',
+              description: 'Alertmanager %(alertmanagerName)s failed to send {{ $value | humanizePercentage }} of notifications to {{ $labels.integration }}.' % $._config,
+            },
+          },
+          {
+            alert: 'AlertmanagerClusterFailedToSendAlerts',
+            expr: |||
+              min by (%(alertmanagerClusterLabels)s) (
+                rate(alertmanager_notifications_failed_total{%(alertmanagerSelector)s}[5m])
+              /
+                rate(alertmanager_notifications_total{%(alertmanagerSelector)s}[5m])
+              )
+              > 0.01
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'All Alertmanager instances in a cluster failed to send notifications.',
+              description: 'The minimum notification failure rate to {{ $labels.integration }} sent from any instance in the %(alertmanagerClusterName)s cluster is {{ $value | humanizePercentage }}.' % $._config,
+            },
+          },
+          {
+            alert: 'AlertmanagerConfigInconsistent',
+            expr: |||
+              count by (%(alertmanagerClusterLabels)s) (
+                count_values by (%(alertmanagerClusterLabels)s) ("config_hash", alertmanager_config_hash{%(alertmanagerSelector)s})
+              ) != 1
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Alertmanager instances within the same cluster have different configurations.',
+              description: 'Alertmanager instances within the %(alertmanagerClusterName)s cluster have different configurations.' % $._config,
+            },
+          },
+          // Both the following critical alerts, AlertmanagerClusterDown and
+          // AlertmanagerClusterCrashlooping, fire if a whole cluster is
+          // unhealthy. It is implied that a generic warning alert is in place
+          // for individual instances being down or crashlooping.
+          {
+            alert: 'AlertmanagerClusterDown',
+            expr: |||
+              max by (%(alertmanagerClusterLabels)s) (
+                avg_over_time(up{%(alertmanagerSelector)s}[5m])
+              ) < 0.5
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'All Alertmanager instances within the same cluster are down.',
+              description: 'Each Alertmanager instances within the %(alertmanagerClusterName)s cluster has been up for less than {{ $value | humanizePercentage }} of the last 5m.' % $._config,
```

Comment:
Should'nt we alert earlier? if they are all down it's too late to get the notification :) as a reminder it is regularly said that one alertmanager cluster is enough company-wide

Reply:
Right. Let me rephrase this to alert if half of the instances are down.

In any case, a proper alerting setup should also have an end-to-end test with an always firing alert, which, if it doesn't call in anymore, will trigger a secondary alert via a service like Dead Man's Snitch.

Reply:
OK, done. The alerts fire now if half or more of the cluster is affected, which means 2 or more with a cluster of 4 instances, which means that you need two failed instances (rather than only one) before getting a page.